classes/class-wpcom-liveblog-wp-cli.php

```diff
@@ -13,67 +13,96 @@ public function readme_for_github() {
 		$readme = $this->markdownify_headings( $readme );
 		echo $readme;
 	}
-	
+
 	/**
 	 * Fix wp_commentmeta table so archived liveblog posts comments display properly.
 	 *
 	 * @subcommand fix-archive
 	*/
-	public function fix_archive() {
+	public function fix_archive( $args, $assoc_args ) {
 		global $wpdb;
 
+		// Grab the dryrun flag from the assoc arguments if its there and define our falg as rwquired.
+		$is_dryrun = ( isset( $assoc_args['dryrun'] ) ) ? true : false;
+
 		// find all liveblogs
+		WP_CLI::line( 'Finding All Live Blog Entries..' );
+
 		$posts = new WP_Query( array(
 									'order'    => 'ASC',
 									'orderby'  => 'ID',
-									'meta_key' => 'liveblog' 
+									'meta_key' => 'liveblog'
 									) );
 
-		foreach( $posts->posts as $post ) {
+		//How many live blogs do we have?
+		$total_liveblogs = count($posts->posts);
+		$current_liveblog = 0;
+
+		//Feedback to the user
+		WP_CLI::line( 'Found ' . $total_liveblogs . ' Live Blogs.' );
+
+		foreach ($posts->posts as $post) {
+
+			//Increment the count so we get a more human readable index, inital value becomes 1 rather than 0.
+			$current_liveblog ++;
+
+			//Tell the user what we are doing, but lets colour this one se we can see its a new Liveblog in the console output.
+			// Note: Assigned to a varibale to be echo'd out, calling WP_CLI::colorize directly didnt output.
+			$intro = WP_CLI::colorize( "%4 Processing Liveblog {$current_liveblog} of {$total_liveblogs} %n");
```

Comment:
[WP_CLI::colorize()](https://make.wordpress.org/cli/handbook/internal-api/wp-cli-colorize/) only makes the string fancy, it doesn't output. This needs to be further wrapped by `WP_CLI::line()` rather than echoing.

Reply:
👍